search/search.php

```diff
@@ -20,4 +20,15 @@
 
 require_once __DIR__ . '/includes/classes/class-search.php';
 
-do_action( 'vip_search_loaded' );
+if ( \Automattic\VIP\Search\Search::are_es_constants_defined() ) {
+	do_action( 'vip_search_loaded' );
```

Comment:
The `do_action()` should actually come at the end, after we've initialized the instance (which is how it worked previously).

Reply:
Make sense, that is better.

However the original implementation did trigger that action before the instance if I understand php code correctly?

The `000-vip-init.php` would run  `require_once( __DIR__ . '/search/search.php' );` which would run the whole `/search/search.php` file including `do_action()` statement. After which the `000-vip-init.php` would continue to setup the instance, correct?

Reply:
Ah crap you're right, my bad. Not sure what I was thinking, long day yesterday 😅

Reply:
I did a search and didn't find anything using this action, so should be perfectly safe to move its location slightly like this - makes more sense to do it after instantiation, which is possible now that we moved this code.